File: src/adapter/festa/fantasia/RoupaDeVampiro.java
package adapter.festa.fantasia;

// Adaptador especializado

public class RoupaDeVampiro extends Fantasia {

    public RoupaDeVampiro(Pessoa pessoa) {
        super(pessoa);
    }

    @Override
    public String nomeDoPersonagem() {
        return "O Vampiro " + pessoa.obterNome();
    }

    @Override
    public String descricaoDaRoupa() {
        return "Aparencia elegante com capa e uma dentadura com dentes afiados";
    }
}

File: src/adapter/festa/fantasia/RoupaDeVampiro.kt
package adapter.festa.fantasia

// Adaptador especializado

class RoupaDeVampiro(pessoa: Pessoa) : Fantasia(pessoa) {

    override fun nomeDoPersonagem(): String {
        return "O Vampiro " + pessoa.obterNome()
    }

    override fun descricaoDaRoupa(): String {
        return "Aparencia elegante com capa e uma dentadura com dentes afiados"
    }
}
